sycl/include/sycl/ext/intel/esimd/math.hpp

```diff
@@ -370,41 +383,64 @@ std::enable_if_t<detail::is_esimd_scalar<T>::value, T>(min)(T src0, T src1,
 #define __ESIMD_EMATH_IEEE_COND                                                \
   detail::is_generic_floating_point_v<T> && (sizeof(T) >= 4)
 
+#define __ESIMD_EMATH_SPIRV_COND                                               \
+  std::is_same_v<T, float> || std::is_same_v<T, sycl::half>
+
 /// Inversion - calculates (1/x). Supports \c half and \c float.
 /// Precision: 1 ULP.
-__ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_COND, inv, inv)
+__ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_SPIRV_COND, inv, recip)
```

Comment:
With this patch `__ESIMD_EMATH_COND` lost all uses. The macro needs to be removed as well (define and undef) Please check other/similar macros if there are such.